### PR TITLE
Array support with the AKV configuration provider

### LINKED ISSUE #8750: Does this configuration provider support arrays?
The configuration for Serilog in the appsettings.json file looks something like this:

```
"Serilog": {
    "MinimumLevel": "Debug",
    "WriteTo": [
      {
        "Name": "AzureTableStorage",
        "Args": {
          "storageTableName": "logs",
          "connectionString": ""
        }
      }
    ]
  }
```

I've been trying to replicate the structure using the '--' delimiter notation so I can store the connection string in Key Vault and inject it at runtime but it doesn't appear to be working. Is it possible to replicate the Serilog configuration object using this provider?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2cc0b77a-df21-7007-5cec-4399d98c6dc6
* Version Independent ID: efe5ae5e-ff64-f1c7-6920-4512fc8f36d1
* Content: [Azure Key Vault configuration provider in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-2.1)
* Content Source: [aspnetcore/security/key-vault-configuration.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/key-vault-configuration.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #8750 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-1.1&branch=pr-en-us-9064)

Uses the explain-by-example strategy based on the issue discussion with @Jaffacakes82. JSON is familiar to devs, so a SxS show-and-tell with the AKV keys might make sense.

@Jaffacakes82, would this have worked for you?

### CODE REVIEW

**Comment by Jaffacakes82** [comment]
@guardrex yep - this would have done the trick.

Thanks.

File: aspnetcore/security/key-vault-configuration.md [hunk 1]
```diff
@@ -57,6 +57,48 @@ When you run the app, a webpage shows the loaded secret values:
...
 
+## Bind an array to a class
+
+The provider is capable of reading configuration values into an array for binding to a POCO array.
+
+When reading from a configuration source that allows keys to contain colon (`:`) separators, a numeric key segment is used to distinguish the keys that make up an array (`:0:`, `:1:`, … `:{n}:`). For more information, see [Configuration: Bind an array to a class](xref:fundamentals/configuration/index#bind-an-array-to-a-class).
+
+Azure Key Vault keys can't use a colon as a separator. The approach described in this topic uses double dashes (`--`) as a separator for hierarchical values (sections). Array keys are stored in Azure Key Vault with double dashes and numeric key segments (`--0--`, `--1--`, … `--{n}--`).
+
+Examine the following [Serilog](https://serilog.net/) JSON file configuration. There are two `WriteTo` sinks specified in a JSON array:
```
Comment (scottaddie) [inline aspnetcore/security/key-vault-configuration.md]: Reword this 2nd sentence. Not sure what's meant by "sinks". Did you mean to say "There are two object literals defined within the `WriteTo` array:"?

**Comment by scottaddie** [comment]
@guardrex Please also update the date on the doc.

**Comment by guardrex** [comment]
Let's see if I try to both clean it up as you suggest + define *sinks*. That way, the reader isn't left scratching their head on what they're looking at.

Otherwise, we might want to use a different example. This looks ok to me tho 🤔.

**Comment by scottaddie** [comment]
@guardrex I'm okay with the sink terminology if it's defined.

**Comment by guardrex** [comment]
@scottaddie Last commit ok? If so, it's *'GO' for throttle up!* :rocket:

File: aspnetcore/security/key-vault-configuration.md [hunk 2]
```diff
@@ -57,6 +57,48 @@ When you run the app, a webpage shows the loaded secret values:
...
 
+## Bind an array to a class
+
+The provider is capable of reading configuration values into an array for binding to a POCO array.
+
+When reading from a configuration source that allows keys to contain colon (`:`) separators, a numeric key segment is used to distinguish the keys that make up an array (`:0:`, `:1:`, … `:{n}:`). For more information, see [Configuration: Bind an array to a class](xref:fundamentals/configuration/index#bind-an-array-to-a-class).
+
+Azure Key Vault keys can't use a colon as a separator. The approach described in this topic uses double dashes (`--`) as a separator for hierarchical values (sections). Array keys are stored in Azure Key Vault with double dashes and numeric key segments (`--0--`, `--1--`, … `--{n}--`).
+
+Examine the following [Serilog](https://serilog.net/) logging provider configuration provided by a JSON file. There are two object literals defined in the `WriteTo` array that reflect two Serilog *sinks*, destinations for logging output:
```
Comment (scottaddie) [inline aspnetcore/security/key-vault-configuration.md]: I recommend an mdash instead of a comma. Other than that, this looks good.

Comment (guardrex) [reply aspnetcore/security/key-vault-configuration.md]: mmmmmm ... 🤔 ... let me try something else first. Stand-by.

**Comment by guardrex** [comment]
I've noticed that you like the 'ole &mdash; mdash ......... me? ...... I'm trying to squeeze a few more &dagger; daggers into the docs! :smile: lol

**Pull request merged**